justfile

```diff
@@ -103,6 +103,16 @@ populate target=default_deploy_target platform=default_platform:
     if [[ -f ./{{ workspace_dir }}/deployment/deployment.yml ]]; then
         echo "---" >> ./{{ workspace_dir }}/deployment/deployment.yml
     fi
+    if [ -n "$CONTRAST_GHCR_READ" ]; then
+        cat > "./{{ workspace_dir }}/contrast-imagepuller.toml" <<EOF
+    [registries]
+    [registries."ghcr.io."]
+    auth = "$(printf "user-not-required-here:%s" "$CONTRAST_GHCR_READ" | base64 -w0)"
+    EOF
+        kubectl create secret generic contrast-node-installer-imagepuller-config \
+            --from-file "contrast-imagepuller.toml"="./{{ workspace_dir }}/contrast-imagepuller.toml" \
+            --namespace {{ target }}${namespace_suffix-}
```

Comment:
This shouldn't happen in populate but in apply (runtime).

Reply:
Makes sense, thx. Moved the writing of the namespace to a separate recipe, secret creation is in apply runtime now.